src/app/battle.service.ts

```diff
@@ -156,8 +156,19 @@ export class BattleService {
         }
       }
       //  Remove death ships
-      playerShips = playerShips.filter(s => s.armor.gt(0));
-      enemyShip = enemyShip.filter(s => s.armor.gt(0));
```

Comment:
`'filter'` create and return a new array so creating one ourselves while freeing 'dead' ship should be a non-issue.